tcms/static/js/utils.js

```diff
@@ -1,17 +1,32 @@
 /*
     Used to update a select when something else changes.
 */
-function updateSelect(data, selector, id_attr, value_attr) {
+function updateSelect(data, selector, id_attr, value_attr, group_attr) {
     var _select_tag = $(selector)[0];
     var new_options = '';
 
-    // in some cases, e.g. TestRun search, the 1st <option> element is ---
+	currentGroup = '';
+	// check if using multiple select
+	isMultiple = _select_tag.attributes.getNamedItem('multiple') !== null;
+    
+	// in some cases using single select, the 1st <option> element is ---
     // which must always be there to indicate nothing selected
-    if (_select_tag.options.length) {
+    if (!isMultiple && _select_tag.options.length) {
         new_options = _select_tag.options[0].outerHTML;
     }
+	
+	data.forEach(function(element) {
+		if(isMultiple) {
+			if (group_attr != null && currentGroup !== element[group_attr]) {
+				if (currentGroup !== '') {
+					// for all but the first time group changes, add a closing optgroup tag
+					new_options += '</optgroup>'
+				}
```

Comment:
I fail to understand why this seems to work but I think the browser is closing missing html tags automatically.

From what I can see when iterating over the last element we should be missing the closing </optgroup> tag b/c on line 30 we'll add the last <option> and then the loop will exit. If a new <optgroup> was started there's nothing afterwards to close it.